### ISSUE TITLE
Error in syndication.py

### ISSUE THREAD

**Issue by zszia** [LukeMurphey/splunk-syndication-input#2]
In line 306 the checkpoint is saved with the key **last_run**

```
if last_entry_date is not None:
    self.save_checkpoint_data(checkpoint_dir, stanza, { 'last_run' : last_run,'last_entry_date' : time.mktime(last_entry_date)})
```

In line 373 you check for key **last_ran**
```
if checkpoint_data is not None and 'last_ran' in checkpoint_data:
```
Maybe I'm wrong but I think, this can never evaluate to True.

**Comment by LukeMurphey** [comment]
I think this can evaluate to true in a particular case (if I am following this correctly).

You can see the call at the bottom to get_non_deviated_last_run().

That call will return a last_run value if the provided value is none. See https://github.com/LukeMurphey/splunk-modular-input/blob/master/src/modular_input_base_class.py#L880

That is where that value would get initialized.

This logic is kinda hard to follow. I put a ton of comments in https://github.com/LukeMurphey/splunk-modular-input/blob/master/src/modular_input_base_class.py#L863 because this timing stuff is not simple and has a bunch of edge cases that have to be handled.
